### PR TITLE
updated check_public_bucket() and check_private_bucket() to accept an optional object_uri, and updated basic bucket check

### PR DESCRIPTION
**Pull request by dpflowersjr**
updated check_public_bucket() and check_private_bucket() to accept an optional object_uri, and updated  basic bucket check

### CODE REVIEW

File: rain_api_core/egress_util.py [hunk 1]
```diff
@@ -150,32 +144,36 @@ def process_request(varargs, b_map):
...
-
+def check_public_bucket(bucket, b_map, optional_uri=""):
     # Check for PUBLIC_BUCKETS in bucket map file
     if 'PUBLIC_BUCKETS' in b_map:
         log.debug('we have a PUBLIC_BUCKETS in the ordinary bucketmap file')
         for pub_bucket in b_map['PUBLIC_BUCKETS']:
-            #log.debug('is {} the same as {}?'.format(bucket, prepend_bucketname(pub_bucket)))
-            if bucket == prepend_bucketname(pub_bucket):
+            # log.debug('is {} the same as {}?'.format(bucket, prepend_bucketname(pub_bucket)))
+            if bucket_prefix_ismatch(bucket, pub_bucket, optional_uri):
```
Comment (bbuechler) [inline rain_api_core/egress_util.py:166]: It looks like you dropped the call to prepend_bucketname(). What happens when buckets have a bucket map prefix now?

Comment (dpflowersjr) [reply rain_api_core/egress_util.py:166]: `Python 3.8.5 (default, Jul 28 2020, 12:59:40) 
[GCC 9.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> bucket_check='bucket1'
>>> bucket_map='bucket1/prefix1/'
>>> object='prefix1/path/object'
>>> if bucket_check == bucket_map.split('/')[0] and object.startswith("/".join(bucket_map.split('/')[1:])):
...     print(True)
... 
True
>>> bucket_check='bucket2'
>>> bucket_map='bucket1/prefix1'
>>> object='prefix1/path/object'
>>> if bucket_check == bucket_map.split('/')[0] and object.startswith("/".join(bucket_map.split('/')[1:])):
...     print(True)
... 
>>> 
`

Comment (dpflowersjr) [reply rain_api_core/egress_util.py:166]: I copied and pasted the test. I got back true as expected for a match and returned nothing as expected for a mismatch with bucket maps with a prefix

Comment (bbuechler) [reply rain_api_core/egress_util.py:166]: Object prefixes, what we’re trying to support, are different from Bucket name prefixes, which is what `prepend_bucketname()` accounted for.

File: rain_api_core/egress_util.py [hunk 2]
```diff
@@ -163,7 +163,7 @@ def check_public_bucket(bucket, b_map, optional_uri=""):
         log.debug('we have a PUBLIC_BUCKETS in the ordinary bucketmap file')
         for pub_bucket in b_map['PUBLIC_BUCKETS']:
             # log.debug('is {} the same as {}?'.format(bucket, prepend_bucketname(pub_bucket)))
-            if bucket_prefix_ismatch(bucket, pub_bucket, optional_uri):
+            if bucket_prefix_ismatch(bucket, pub_bucket, optional_uri) and bucket == prepend_bucketname(pub_bucket):
```
Comment (bbuechler) [inline rain_api_core/egress_util.py:166]: Will this return `True` if `BUCKETNAME_PREFIX='prod'` and the bucket_map is: 
```
MAP:
    product: productbucket
PUBLIC_BUCKET:
   productbucket/browse: "this is public data"
```
...and a request comes in for
`https://<tea>/product/browse/image/granule.png`?

What are the for `bucket`, `pub_bucket`, `optional_uri`, and the return value from `prepend_bucketname(pub_bucket)`?

File: rain_api_core/egress_util.py [hunk 3]
```diff
@@ -150,32 +144,35 @@ def process_request(varargs, b_map):
...
                 log.debug('found a public, we\'ll take it')
                 return True
 
     # Did not find this in public bucket list
     log.debug('we did not find a public bucket for {}'.format(bucket))
     return False
+
+
+def bucket_prefix_ismatch(bucket_check, bucket_map, optional_uri):
+    bucket = prepend_bucketname(bucket_check)
```
Comment (bbuechler) [inline rain_api_core/egress_util.py:177]: Is this the right one to prefix?

Comment (dpflowersjr) [reply rain_api_core/egress_util.py:177]: Are you suggesting that the public bucket doesn't need to be pre-fixed?

**Pull request closed without merging**